File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.cpp
class Solution {
public:
    string reverseWords(string s) {
        int spaceCnt = 0;
        int left = 0, right;
        for(int i = 0; i < s.size(); i++){
            if(s[i + 1] == ' ' || i == s.size() - 1){
                right = i;
                while(left < right){
                    char tmp = s[left];
                    s[left] = s[right];
                    s[right] = tmp;
                    ++left,--right;
                }
                left = i + 2;
            }
        }
        return s;

        //  ==============================  //
        //  https://leetcode.com/problems/reverse-words-in-a-string-iii/solutions/4111458/91-55-easy-solution-reverse-join
        // stringstream ss(s);
        // string temp,ans;
        // while(ss>>temp){
        //     reverse(temp.begin(),temp.end());
        //     ans+= (temp+" ");
        // }
        
        // return ans.substr(0,ans.size()-1);
    }
};

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.cs
public class Solution {
    public string ReverseWords(string s) {
        //  https://leetcode.com/problems/reverse-words-in-a-string-iii/solutions/4111458/91-55-easy-solution-reverse-join
        string[] words = s.Split();
        string[] reverseWords = new string[words.Length];
        for(int i = 0; i < words.Length; i++){
            char[] wordChars = words[i].ToCharArray();
            Array.Reverse(wordChars);
            reverseWords[i] = new string(wordChars);
        }
        //  https://learn.microsoft.com/ko-kr/dotnet/api/system.string.join?view=net-8.0
        return string.Join(" ", reverseWords);
    }
}